src/linker/Linker.Steps/ValidateVirtualMethodAnnotationsStep.cs

```diff
@@ -40,12 +40,24 @@ protected override void Process ()
 		void ValidateMethodRequiresUnreferencedCodeAreSame (MethodDefinition method, MethodDefinition baseMethod)
 		{
 			var annotations = Context.Annotations;
-			if (annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (method) !=
-				annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (baseMethod))
-				Context.LogWarning (
-					$"Presence of 'RequiresUnreferencedCodeAttribute' on method '{method.GetDisplayName ()}' doesn't match overridden method '{baseMethod.GetDisplayName ()}'. " +
-					$"All overridden methods must have 'RequiresUnreferencedCodeAttribute'.",
-					2046, method, subcategory: MessageSubCategory.TrimAnalysis);
+			bool methodHasAttribute = annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (method);
+			if (methodHasAttribute != annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (baseMethod)) {
+				string message = string.Empty;
+				string var0 = nameof (RequiresUnreferencedCodeAttribute);
+				string var1 = method.GetDisplayName ();
+				string var2 = baseMethod.GetDisplayName ();
+				if (!methodHasAttribute && !baseMethod.DeclaringType.IsInterface)
```

Comment:
This code looks identical. Let's share it.

Reply:
SharedStrings is in the limbo between two contexts Mono.Linker and ILlink.RoslynAnalyzer, I had two options to share it:
- Make ILLink.Shared a project
- Create a resource using reflection

I implemented the second only because I feel like the push is to not have an additional project, although I would prefer an additional project over the risk of getting things based on the executing assembly.

Reply:
> Looks like the tests still need to move to a base class

I didn't create a base class since after #2077 these tests won't need a base class anymore. When I implementing #2077 I will create a base class to share everything left in the RequiresUnreferencedCodeAnalyzerTests and RequiresAssemblyFilesTests